Packet++/header/TcpReassembly.h

```diff
@@ -106,10 +107,14 @@ namespace pcpp
 		uint16_t dstPort;
 		/** A 4-byte hash key representing the connection */
 		uint32_t flowKey;
-		/** Start TimeStamp of the connection */
+		/** Start timestamp of the connection with microsecond precision */
 		timeval startTime;
-		/** End TimeStamp of the connection */
+		/** End timestamp of the connection with microsecond precision */
 		timeval endTime;
+		/** Start timestamp of the connection with nanosecond precision */
+		std::chrono::time_point<std::chrono::system_clock> startTimePrecise;
```

Comment:
It's not guarantee that std::chrono::time_point<std::chrono::system_clock> have nano seconds resolution.
It depends on Clock::duration. On macOS, system_clock::duration is microseconds.

IMO i would change all the timeval to timespec which is C11 posix and better defined.

Reply:
Tbh, using timespec might be simpler, as the timepoints in the chrono library are more specialized to be used with a clock, and we get the time externally anyway to we don't really have one.

Reply:
Please see: https://github.com/seladb/PcapPlusPlus/pull/1591#discussion_r1782353418

Reply:
I meant we whould explicit the duration like this:

```
using time_point_ns = std::chrono::time_point<std::chrono::high_resolution_clock, std::chrono::nanoseconds>;
time_point_ns startTimePrecise;
time_point_ns endTimePrecise;

```

This will also work for std::chrono::system_clock

Reply:
Why do we need to be explicit? If we do:
```cpp
std::chrono::time_point<std::chrono::high_resolution_clock>
```

Then the user can decide which precision they want, for example, if they want nano sec they can do the following:
```
std::chrono::duration_cast<std::chrono::nanoseconds>(stats.begin()>second.connData.startTimePrecise.time_since_epoch())
```

But they can also get any other precision